addons/website_slides/controllers/main.py

```diff
@@ -368,3 +377,104 @@ def slides_embed(self, slide_id, page="1", **kw):
         except AccessError: # TODO : please, make it clean one day, or find another secure way to detect
                             # if the slide can be embedded, and properly display the error message.
             return request.render('website_slides.embed_slide_forbidden', {})
+
+    # Profile
+    # ---------------------------------------------------
+    def _prepare_open_slide_user(self, user):
+        slides = request.env['slide.slide'].sudo().search([('user_ids', '=', user.id)])
+        courses = request.env['slide.channel.partner'].sudo().search([('partner_id', '=', user.partner_id.id)]).channel_id
+        courses_completed = courses.filtered(lambda c: c.completed)
+        courses_ongoing = courses - courses_completed
+
+        values = {
+            'uid': request.env.user.id,
+            'user': user,
+            'main_object': user,
+            'slides': slides,
+            'count_slides': len(slides),
+            'courses': courses,
+            'count_courses': len(courses),
+            'is_profile_page': True,
+        }
+        return values
+
+    @http.route(['/slides/user/<int:user_id>'], type='http', auth="public", website=True)
+    def open_cross_slide_user(self, user_id=0, **post):
+        user = request.env['res.users'].sudo().browse([user_id])
```

Comment:
It should use the same _check method as already discussed a bit in previous commit about "who can see a user". We hould have one check point before going into user details.